sync-core/src/main/java/com/cloudant/sync/datastore/DocumentRevisionBuilder.java

```diff
@@ -211,6 +222,21 @@ public MutableDocumentRevision buildMutable() {
     }
 
     /**
+     * Builds and returns the {@link com.cloudant.sync.datastore.ProjectedDocumentRevision} for this builder.
+     * @return {@link com.cloudant.sync.datastore.ProjectedDocumentRevision} for this builder.
+     */
+    public ProjectedDocumentRevision buildProjected() {
+        BasicDocumentRevision.BasicDocumentRevisionOptions options = new BasicDocumentRevision.BasicDocumentRevisionOptions();
+        options.sequence = sequence;
+        options.docInternalId = docInternalId;
+        options.deleted = deleted;
+        options.current = current;
+        options.parent = parent;
+        options.attachments = attachments;
+        return new ProjectedDocumentRevision(docId, revId, body, options, datastore);
+    }
```

Comment:
`DocumentRevisionBuilder` is public. Building a projected document revision shouldn't be a public operation. I don't think updating the public API of the library for this is warranted.

I see that mutable document revisions are also created by this class. I don't think that's a good thing either, but we can remove that later.